### PR TITLE
ExprBlocks - update patterns to include locations, cuboid and chunks

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
This PR aims to update the ExprBlocks expression. 
A few things:
- Use location expressions rather than blocks. This makes for easier syntax. This will not break current scripts as using a block will compare to a location
- Split (between|within), so between stays as a straight line and within returns a cuboid.
- Add a pattern for getting all blocks in a chunk.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** #3021 (I feel like there is another issue regarding this, but I cant find it)

### CODE REVIEW

**Comment by Anarchick** [comment]
> **Related Issues:** #3021 (I feel like there is another issue regarding this, but I cant find it)
This ? #3243

File: src/main/java/ch/njol/skript/expressions/ExprBlocks.java [hunk 1]
```diff
@@ -163,10 +185,15 @@ public boolean isSingle() {
 	
 	@Override
 	public String toString(final @Nullable Event e, final boolean debug) {
-		final Expression<Block> end = this.end;
-		if (end != null) {
+		final Expression<Location> end = this.end;
+		if (chunk != null)
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprBlocks.java:189]: Nitpick but can you put curly braces on all the if / else if statements here ;D

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
very minor

File: src/main/java/ch/njol/skript/expressions/ExprBlocks.java (hunk 1 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprBlocks.java]: done 👍

**Review by APickledWalrus — APPROVED** [review]
LGTM

**Pull request merged**